tests/src/system/basic/WskActionTests.scala

```diff
@@ -0,0 +1,268 @@
+        (wp, assetHelper) =>
+            val name = "helloSyncDoneTwice"
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) => action.create(name, Some(TestUtils.getTestActionFilename("helloSyncDoneTwice.js")))
+            }
+
+            val run = wsk.action.invoke(name, Map("payload" -> testString.toJson))
+            withActivation(wsk.activation, run) {
+                activation =>
+                    activation.response.status shouldBe "success"
+                    activation.response.result shouldBe Some(testResult)
+                    activation.logs.get.mkString(" ") should include(testString)
+            }
+    }
+
+    it should "return the value of the first asynchronous whisk.done()" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val name = "helloSyncDoneTwice"
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) => action.create(name, Some(TestUtils.getTestActionFilename("helloAsyncDoneTwice.js")))
+            }
+
+            val run = wsk.action.invoke(name, Map("payload" -> testString.toJson))
+            withActivation(wsk.activation, run) {
+                activation =>
+                    activation.response.status shouldBe "success"
+                    activation.response.result shouldBe Some(testResult)
+                    activation.logs.get.mkString(" ") should include(testString)
+            }
+    }
+
+    it should "reject an invoke with the wrong parameters set" in {
+        val payload = "bob"
+        val rr = wsk.cli(Seq("action", "invoke", "/whisk.system/samples/helloWorld", payload), expectedExitCode = TestUtils.ANY_ERROR_EXIT)
+        rr.exitCode shouldBe 1
+        rr.stderr should include("Run 'wsk --help' for usage.")
+        rr.stderr should include(s"error: Invalid argument(s): $payload")
+    }
+
+    it should "not be able to use 'ping' in an action" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val name = "ping"
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) => action.create(name, Some(TestUtils.getTestActionFilename("ping.js")))
+            }
+
+            val run = wsk.action.invoke(name, Map("payload" -> "google.com".toJson))
+            withActivation(wsk.activation, run) {
+                activation =>
+                    activation.response.result shouldBe Some(JsObject(
+                        "stderr" -> "ping: icmp open socket: Operation not permitted\n".toJson,
+                        "stdout" -> "".toJson))
+            }
+    }
+
+    ignore should "support UTF-8 as input and output format" in withAssetCleaner(wskprops) {
+        (wp, assetHelper) =>
+            val name = "utf8Test"
+            assetHelper.withCleaner(wsk.action, name) {
+                (action, _) => action.create(name, Some(TestUtils.getCatalogFilename("samples/hello.js")))
```

Comment:
Since I am refactoring all the test cases to remove the catalog, could you change getCatalogFilename into getTestActionFilename?